### PR TITLE
fix(api-client): default redirectURI and prefix relative redirectURIs with activeServer

### LINKED ISSUE #3699: oauth redirect URI should default to current URL
the default redirect URL should be the current URL I think, because we want to have the access code in the reference/client, not on some other domain

### LINKED ISSUE #3700: authorize button doesn‘t work with relative URLs
oauth2 authorize button doesn‘t work with relative URLs (/oauth/authorize), they should use currently selected server as an URL

### PR DESCRIPTION
**Pull request by amritk**
fixes #3699 
fixes #3700

### CODE REVIEW

File: packages/api-client/src/views/Request/libs/oauth2.ts [hunk 1]
```diff
@@ -52,9 +55,25 @@ export const authorizeOauth2 = (
       else if (scheme.flow.type === 'authorizationCode')
         url.searchParams.set('response_type', 'code')
 
+      // Handle relative redirect uris
+      if (scheme.flow['x-scalar-redirect-uri'].startsWith('/')) {
```
Comment (hanspagel) [inline packages/api-client/src/views/Request/libs/oauth2.ts:59]: Let me add docs for x-scalar-redirect-uri tomorrow ✌️

Comment (hanspagel) [reply packages/api-client/src/views/Request/libs/oauth2.ts:59]: Why do we need it?

Comment (amritk) [reply packages/api-client/src/views/Request/libs/oauth2.ts:59]: why do we need to check for the `/` or why do we need x-scalar-redirect-uri?

Comment (amritk) [reply packages/api-client/src/views/Request/libs/oauth2.ts:59]: if its the latter, lets discuss

Comment (hanspagel) [reply packages/api-client/src/views/Request/libs/oauth2.ts:59]: The latter, why do we need the custom attribute and can’t just use the existing redirectUri attribute?

Comment (amritk) [reply packages/api-client/src/views/Request/libs/oauth2.ts:59]: Can tag @geoffgscott but this is something I would like to discuss in person

**Pull request merged**